subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/ToBeFixedForInstantExecutionRule.groovy

```diff
@@ -32,10 +32,16 @@ class ToBeFixedForInstantExecutionRule implements TestRule {
         if (!GradleContextualExecuter.isInstant() || annotation == null) {
             return base
         }
-        if (annotation.value() == ToBeFixedForInstantExecution.Skip.DO_NOT_SKIP) {
-            return new ExpectingFailureRuleStatement(base)
+        ToBeFixedForInstantExecution.Skip skip = annotation.value()
+        String[] bottomSpecs = annotation.bottomSpecs()
+        if (bottomSpecs.length == 0 || bottomSpecs.any { description.className.endsWith(it) }) {
```

Comment:
⭕️ `endsWith(it)` seems too fragile, maybe include a leading dot to avoid requesting the fully qualified bottom spec name? I mean something like `endsWith(".$it")`?